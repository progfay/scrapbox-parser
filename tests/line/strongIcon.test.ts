/* global describe it expect */

describe('strongIcon', () => {
  it('Simple root strong icon', () => {
    expect('[[/icons/+1.icon]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongIcon',
            pathType: 'root',
            path: '/icons/+1'
          }
        ]
      }
    ])
  })

  it('Simple relative strong icon', () => {
    expect('[[me.icon]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongIcon',
            pathType: 'relative',
            path: 'me'
          }
        ]
      }
    ])
  })

  it('Multiple icons', () => {
    expect('[[me.icon*3]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'strongIcon',
            pathType: 'relative',
            path: 'me'
          },
          {
            type: 'strongIcon',
            pathType: 'relative',
            path: 'me'
          },
          {
            type: 'strongIcon',
            pathType: 'relative',
            path: 'me'
          }
        ]
      }
    ])
  })
})
