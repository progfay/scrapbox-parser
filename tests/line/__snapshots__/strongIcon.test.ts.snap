// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`strongIcon Multiple icons 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "me",
        "pathType": "relative",
        "raw": "[[me.icon*3]]",
        "type": "strongIcon",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "raw": "[[me.icon*3]]",
        "type": "strongIcon",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "raw": "[[me.icon*3]]",
        "type": "strongIcon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongIcon Simple relative strong icon 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "me",
        "pathType": "relative",
        "raw": "[[me.icon]]",
        "type": "strongIcon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongIcon Simple root strong icon 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "/icons/+1",
        "pathType": "root",
        "raw": "[[/icons/+1.icon]]",
        "type": "strongIcon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`strongIcon Strong icon and internal link on same line 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "Internal link",
        "pathType": "relative",
        "raw": "[Internal link]",
        "type": "link",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "raw": "[[me.icon]]",
        "type": "strongIcon",
      },
    ],
    "type": "line",
  },
]
`;
